### PR TITLE
Add defaults to pod affinity args

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add defaults and validation to pod affinity args. This is a first step towards removing dependencies in `.hardPodAffinitySymmetricWeight` from the component config.

Also removing pod affinity parameter from algorithm provider registry. It was the only parameter affecting the registry, making the migration harder.

**Which issue(s) this PR fixes**:

Part of #87657, #87703 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/interpodaffinity/plugin.go [hunk 1]
```diff
@@ -55,14 +61,30 @@ func New(plArgs *runtime.Unknown, h framework.FrameworkHandle) (framework.Plugin
...
+	}
+	return pl, nil
+}
+
+func validateArgs(args *Args) error {
+	if args.HardPodAffinityWeight == nil {
+		return nil
+	}
+
+	if *args.HardPodAffinityWeight < minHardPodAffinityWeight || *args.HardPodAffinityWeight > maxHardPodAffinityWeight {
```
Comment (draveness) [inline pkg/scheduler/framework/plugins/interpodaffinity/plugin.go]: nit:

```
weight := *args.HardPodAffinityWeight
if weight < minHardPodAffinityWeight ... {
}
```

File: pkg/scheduler/factory_test.go [hunk 2]
```diff
@@ -190,17 +192,29 @@ func TestCreateFromConfigWithHardPodAffinitySymmetricWeight(t *testing.T) {
 		"priorities" : [
 			{"name" : "RackSpread", "weight" : 3, "argument" : {"serviceAntiAffinity" : {"label" : "rack"}}},
 			{"name" : "NodeAffinityPriority", "weight" : 2},
-			{"name" : "ImageLocalityPriority", "weight" : 1}
+			{"name" : "ImageLocalityPriority", "weight" : 1},
+			{"name" : "InterPodAffinityPriority", "weight" : 1}
```
Comment (draveness) [inline pkg/scheduler/factory_test.go]: What is the prioritiby between `c. hardPodAffinitySymmetricWeight` and `hardPodAffinityWeight` in plugin config, which one would be overwritten if we have them both?

**Review by draveness — CHANGES_REQUESTED** [review]

File: pkg/scheduler/factory_test.go (hunk 2 above)
Comment (alculquicondor) [reply pkg/scheduler/factory_test.go]: It was never documented. But looking at the appends in `Configurator.createFromConfig` and `Configurator.createFromProvider` and how the arrays are processed in `NewFramework`, it looks like the pluginConfig had higher preference. This doesn't change with this PR.

But ultimately, it doesn't matter much, as a follow up PR is removing `c.hardPodAffinitySymmetricWeight` from v1alpha2.

File: pkg/scheduler/framework/plugins/interpodaffinity/plugin.go [hunk 3]
```diff
@@ -55,14 +61,30 @@ func New(plArgs *runtime.Unknown, h framework.FrameworkHandle) (framework.Plugin
...
+		return fmt.Errorf("invalid args.hadPodAffinityWeight: %d, must be in the range %d-%d", *args.HardPodAffinityWeight, minHardPodAffinityWeight, maxHardPodAffinityWeight)
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/scheduler/framework/plugins/interpodaffinity/plugin.go]: good catch. Done

**Comment by ahg-g** [comment]
/lgtm

We need to think about how we make those configurations more accessible to the users. For example should we place the Args of different in-tree plugins in the ```apis/config``` package since they are practically part of the API?

**Comment by alculquicondor** [comment]
/hold cancel

I also opened #87784 for discussing the idea mentioned above.

**Comment by ahg-g** [comment]
/hold

One useful thing enabled by the original approach was defining PluginConfig as part of the provider configuration. Which was also useful for CA when they create a framework with default configuration.

**Comment by alculquicondor** [comment]
> One useful thing enabled by the original approach was defining PluginConfig as part of the provider configuration. Which was also useful for CA when they create a framework with default configuration.

The outcome doesn't change. If they use the default configuration, which doesn't contain any plugin config, the default is set during plugin instantiation.

**Comment by ahg-g** [comment]
> > One useful thing enabled by the original approach was defining PluginConfig as part of the provider configuration. Which was also useful for CA when they create a framework with default configuration.
> 
> The outcome doesn't change. If they use the default configuration, which doesn't contain any plugin config, the default is set during plugin instantiation.

Default provider did contain a plugin config, although it was for the inter-pod affinity score function, which does not impact CA, but whose to say this will not change in the future?

I also don't see why creating the plugin config in factory.go is better than baking it in the provider registry?

**Comment by alculquicondor** [comment]
> Default provider did contain a plugin config, although it was for the inter-pod affinity score function, which does not impact CA, but whose to say this will not change in the future?

Exactly, so this plugin config is now set as part of plugin creation.

> I also don't see why creating the plugin config in factory.go is better than baking it in the provider registry?

It is not. But this is going away in my next PR, when I remove the parameter from the component config.

**Comment by ahg-g** [comment]
> > Default provider did contain a plugin config, although it was for the inter-pod affinity score function, which does not impact CA, but whose to say this will not change in the future?
> 
> Exactly, so this plugin config is now set as part of plugin creation.
> 
> > I also don't see why creating the plugin config in factory.go is better than baking it in the provider registry?
> 
> It is not. But this is going away in my next PR, when I remove the parameter from the component config.

I am not sure I understand how this is going to go away (and what do you mean this is part of plugin creation). Don't we need to define the default plugin config somewhere? is it possible to look at the next PR now?

**Comment by ahg-g** [comment]
/hold cancel

Discussed offline, defaults will be backed into the plugin itself.

**Pull request merged**